plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/AddFilesProcedure.java

```diff
@@ -0,0 +1,403 @@
+
+            log.debug("Start new transaction");
+            Transaction transaction = table.newTransaction();
+            if (!table.properties().containsKey(DEFAULT_NAME_MAPPING)) {
+                log.debug("Update default name mapping property");
+                transaction.updateProperties()
+                        .set(DEFAULT_NAME_MAPPING, toJson(nameMapping))
+                        .commit();
+            }
+            List<DataFile> dataFiles = dataFilesBuilder.build();
+            log.debug("Append data %d data files", dataFiles.size());
+            AppendFiles appendFiles = isMergeManifestsOnWrite(session) ? transaction.newAppend() : transaction.newFastAppend();
+            dataFiles.forEach(appendFiles::appendFile);
+            appendFiles.commit();
+            transaction.commitTransaction();
+            log.debug("Successfully added files to %s table", targetName);
+        }
+        catch (Exception e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to add files: " + firstNonNull(e.getMessage(), e), e);
+        }
+    }
+
+    private static List<DataFile> buildDataFiles(
+            TrinoFileSystem fileSystem,
+            SourceType sourceType,
+            RecursiveDirectory recursive,
+            HiveStorageFormat format,
+            String location,
+            PartitionSpec partitionSpec,
+            Optional<StructLike> partition,
+            Schema schema,
+            Set<Integer> requiredFields)
+            throws IOException
+    {
+        List<DataFile> dataFiles = switch (sourceType) {
+            case TABLE -> MigrationUtils.buildDataFiles(fileSystem, recursive, format, location, partitionSpec, partition, schema);
+            case LOCATION -> buildDataFilesFromLocation(fileSystem, recursive, format, location, partitionSpec, partition, schema);
+        };
+
+        if (!requiredFields.isEmpty()) {
+            for (DataFile dataFile : dataFiles) {
+                Map<Integer, Long> nullValueCounts = dataFile.nullValueCounts();
+                for (Integer field : requiredFields) {
+                    if (nullValueCounts.get(field) > 0) {
+                        throw new TrinoException(CONSTRAINT_VIOLATION, "NULL value not allowed for NOT NULL column: " + schema.findField(field).name());
+                    }
+                }
+            }
+        }
+
+        return dataFiles;
+    }
+
+    private static List<DataFile> buildDataFilesFromLocation(
+            TrinoFileSystem fileSystem,
+            RecursiveDirectory recursive,
+            HiveStorageFormat format,
+            String location,
+            PartitionSpec partitionSpec,
+            Optional<StructLike> partition,
+            Schema schema)
+            throws IOException
+    {
+        if (fileSystem.directoryExists(Location.of(location)).orElse(false)) {
+            return MigrationUtils.buildDataFiles(fileSystem, recursive, format, location, partitionSpec, partition, schema);
```

Comment:
MigrationUtils is redundant

Reply:
Nope, the same method name exists in this class.